elyra/pipeline/pipeline_definition.py

```diff
@@ -432,6 +436,23 @@ def validate(self) -> list:
 
         return validation_issues
 
+    def propagate_global_properties(self):
+        """
+        For any global pipeline properties set (e.g. runtime image, volume), propagate
+        the values to any nodes that do not set their own value for that property.
+        """
+        global_properties = self.primary_pipeline.get_property("globals", {})
+        for global_prop, global_value in global_properties.items():
+            if not global_value:
+                continue
+
+            for pipeline in self.pipelines:
+                for node in pipeline.nodes:
+                    if Operation.is_generic_operation(node.op):
+                        node_value = node.get_component_parameter(global_prop)
+                        if not node_value:
+                            node.set_component_parameter(global_prop, global_value)
```

Comment:
This needs to coerce list-valued properties (i.e., envs) where both sources of values may be in play.  For those, I would suggest converting each list to a set, then taking the union of the two (I believe there's an option, or ordering, that allows you to _prefer_ the values from the node over the globals), then convert the result back into a list.

(Hmm, I can't recall what the decision was in this case when both have values.  Seems like it would defeat the purpose of globals for envs if they aren't merged.)

Reply:
Good point! I also can't recall the decision here, though

Reply:
for lists (e.g. env vars) we were going to merge and when duplicated, node value has precendence.

Reply:
What if a variable is defined globally (`KEY=VAL`) and locally (`KEY=`). Is `<empty>` string the node value or is that treated as not set?

Reply:
I think we need to be consistent and honor the node's value, so this would work just like today if `KEY=` is set on the node.

Reply:
I agree, but I believe this might result in an issue because of how auto-detect works. Let's use the following simplified scenario:
- Pipeline default env variable `KEY` is set to `VALUE`
- Notebook contains code `os.environ['KEY'] = 123` that causes auto-detect to add `KEY=` to the node's list of environment variables
- If the auto-added local `KEY=` overrides the pipeline default, it might not be in the users' interest

(I am using just one variable here, but we need to consider pipelines with many notebooks and variables)

Reply:
That makes sense.  So "non-empty" node _values_ take precedence, otherwise, the pipeline property flows.